tests/run_demo.py

```diff
@@ -38,18 +41,15 @@ def default(self, o):
 AGENT_DEFENDER = "defender"
 
 lang_file = "tests/org.mal-lang.coreLang-1.0.0.mar"
-lang_spec = specification.load_language_specification_from_mar(lang_file)
-specification.save_language_specification_to_json(lang_spec, "lang_spec.json")
-lang_classes_factory = LanguageClassesFactory(lang_spec)
-lang_classes_factory.create_classes()
 
-lang_graph = LanguageGraph(lang_spec)
+lang_graph = LanguageGraph.from_mar_archive(lang_file)
+lang_classes_factory = LanguageClassesFactory(lang_graph)
+lang_classes_factory.create_classes()
 
-model = Model("Test Model", lang_spec, lang_classes_factory)
-model.load_from_file("tests/example_model.json")
+model = Model.load_from_file("tests/example_model.yml", lang_classes_factory)
 
-attack_graph = AttackGraph(lang_spec, model)
-attack_graph.attach_attackers(model)
+attack_graph = AttackGraph(lang_graph, model)
+attack_graph.attach_attackers()
 attack_graph.save_to_file("tmp/attack_graph.json")
```

Comment:
I think this should be further simplified:

```
model = Model.load_from_file("tests/example_model.yml", lang_file)
attack_graph = AttackGraph(lang_file, model)
attack_graph.attach_attackers()
```

In other words, the language factory and the language graph should not really be exposed to the users unless they need sth to do with them. It makes the API more consistent too, since the first argument of the model constructor is already a path to a file and not an object.

This is also relevant to this issue: https://github.com/mal-lang/mal-toolbox/issues/21.